packages/next-auth/src/providers/asgardeo.ts

```diff
@@ -0,0 +1,34 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface AsgardeoProfile {
+  sub: string
+  name: string
+  email: string
+}
+
+export default function Asgardeo<P extends Record<string, any> = AsgardeoProfile>(
+  options: OAuthUserConfig<P> & {
+    organization: string
+    scopes?: string
+  }
+): OAuthConfig<P> {
+  return {
+    id: "asgardeo",
+    name: "Asgardeo",
+    type: "oauth",
+    wellKnown: `https://api.asgardeo.io/t/${options?.organization}/oauth2/token/.well-known/openid-configuration`,
```

Comment:
```suggestion
    wellKnown: `${options.issuer}/oauth2/token/.well-known/openid-configuration`,
```

I'm not really sure why the well-known endpoint is under /oauth2/token, but this seems to be correct after what I could find in:

https://github.com/asgardeo/asgardeo-auth-spa-sdk/blob/6d11d2627d0fe6bd3c1dc48b327b6c5dc42c1a62/README.md#the-authclientconfig-interface

You could potentially take it up with them, since I think this is not spec-compliant. According to the spec:

> OpenID Providers supporting Discovery MUST make a JSON document available at the path formed by concatenating the string /.well-known/openid-configuration to the Issuer

https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig1

So I think this should be:
`${options.issuer}/.well-known/openid-configuration`

Where `options.issuer` is `https://api.asgardeo.io/t/organization`